src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -628,28 +630,40 @@ void NetworkOPsImp::setStateTimer ()
 
 void NetworkOPsImp::setHeartbeatTimer ()
 {
-    m_heartbeatTimer.setExpiration (LEDGER_GRANULARITY);
+    // Only start the timer if waitHandlerCounter_ is not yet joined.
+    if (auto optionalCountedHandler = waitHandlerCounter_.wrap (
+        [this] (boost::system::error_code const& e)
+        {
+            if ((e.value() == boost::system::errc::success) &&
```

Comment:
In the event of a failure, we may want to log the error and have some recourse to resetting the timer.